python/ray/tune/search/basic_variant.py

```diff
@@ -343,10 +356,25 @@ def add_configurations(
                     "dimensions/size of the provided grid search."
                 )
 
-            previous_samples = self._total_samples
+            # Find all unresolved values that potentially need to be resolved later.
+            # Note that we only handle references that are part of the config dict.
+            unresolved_configs = _unresolved_values(experiment.spec["config"])
+            for k, v in unresolved_configs.items():
+                # TODO(junogng) : handle Function type as well.
```

Comment:
Just for my own understanding, the `Function` case is concerning `tune.sample_from(lambda spec: ...)`, and what needs to be changed is updating the spec that gets passed into the lambda function?

Reply:
see the new implementation.